src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -796,4 +801,102 @@ void deleteUserDataWhenUserNotFound() {
         verify(userService, never()).delete(userId);
         verify(roleService, never()).delete(any());
     }
+
+    @Test
+    void createIdentityData() throws DataException {
+        IdentitySpec identitySpec = new IdentitySpec();
+        identitySpec.setUserId("userId");
+        identitySpec.setEmail("email@example.com");
+        identitySpec.setVerificationSource("source");
+
+        IdentityData mockIdentityData = new IdentityData("identityId");
+
+        when(identityService.create(identitySpec)).thenReturn(mockIdentityData);
+
+        IdentityData createdIdentityData = testDataService.createIdentityData(identitySpec);
+
+        assertEquals("identityId", createdIdentityData.getId());
```

Comment:
Could we check the object rather than just its id field?
```suggestion
        assertEquals(mockIdentityData, createdIdentityData);
```

Reply:
updated as suggested

Reply:
I actually meant to assert the `IdentityData` is equals to the one returned by the identityService, not just its id?

Reply:
asserted with mockIdentityData and createdIdentityData